src/main/java/io/aiven/kafka/auth/json/JsonUtils.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2019 Aiven, Helsinki, Finland. https://aiven.io/
+ */
+
+package io.aiven.kafka.auth.json;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+public class JsonUtils {
+
+    private static final GsonBuilder GSON_FACTORY =
+        new GsonBuilder()
+            .registerTypeAdapter(Pattern.class, new RegexpJsonDeserializer());
+
+    public static List<AivenAcl> readAclConfiguration(final Path configFile)
```

Comment:
I feel a bit uncomfortable putting `readAclConfiguration`, `readPrincipalMappings`, and `readUserPassword` into a utils class: they have their domains where they belong and spreading this logic may not be really justifiable.

I'd suggest moving these funtions into the relevant classes and make some `JsonReader` class out of the rest.

Reply:
the purpose of the class is to exclude copy/paste solution which we had. 
What do you think about introducing JsonReader iface and subclasses with Factory?

Reply:
I'm thinking about something like:
- some unified way to read JSON, e.g. based on the path and the type -- this can be a static class or whatever; and
- each class that reads JSON uses this way but keeping their stuff internal.

The static class with a single method seems pretty fine here, but feel free to do whatever class hierarchy you deem necessary.